src/Query/Builder.php

```diff
@@ -1084,7 +1086,7 @@ protected function compileWhereBasic(array $where): array
             $operator = $operator === 'regex' ? '=' : 'not';
         }
 
-        if (! isset($operator) || $operator == '=') {
+        if (! isset($operator) || $operator === '=' || $operator === 'eq') {
```

Comment:
This could result in a subtle behavioral change when using `MongoDB\BSON\Regex` objects. Quoting [`$eq` docs](https://www.mongodb.com/docs/manual/reference/operator/query/eq/):

> Specifying the [$eq](https://www.mongodb.com/docs/manual/reference/operator/query/eq/#mongodb-query-op.-eq)
 operator is equivalent to using the form { field: <value> } except when the <value> is a regular expression. See below for [examples.](https://www.mongodb.com/docs/manual/reference/operator/query/eq/#std-label-eq-usage-examples)

Reply:
`{ field: { eq: Regex } }` is not supported ATM as it will always end to `{ field: Regex }` for matching the regex. I don't think storing Regex in database is common enough to fix it know. [PHPORM-74](https://jira.mongodb.org/browse/PHPORM-74)